src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs

```diff
@@ -725,8 +718,7 @@ internal void FinishOperationSuccess(SocketError socketError, int bytesTransferr
                     // Mark socket connected.
                     if (socketError == SocketError.Success)
                     {
-                        if (s_loggingEnabled)
-                            SocketsEventSource.Connected(_currentSocket, _currentSocket.LocalEndPoint, _currentSocket.RemoteEndPoint);
+                        NetEventSource.Connected(_currentSocket, _currentSocket.LocalEndPoint, _currentSocket.RemoteEndPoint);
```

Comment:
Ok. While I was hoping to avoid all those checks, I get that having them makes it easier to always do the right thing and avoid mistakes. I'll add a commit that adds guards everywhere.